## Import Packages

In [1]:
import pandas as pd

## Part 1: Cost and Revenues

### A

In [2]:
# 100 one-time TV mounting bookings GTV
one_time_gtv = 100 * 130

# Customers with recurring service plans
recurring_gtv = 30 * (130 - (130 * .2))  # or 30 * (130 * (.8))

# TV mounting bookings through retails partners are $150 (retail partner keeps 50%) - $75 per booking to Angi
# Total shown to customer
retail_partner_gtv = (45 * 150)

# Print results
print(f'Total GTV for 100 one-time mounting bookings = ${one_time_gtv}')
print(f'Total GTV for 30 handyman service bookings = ${recurring_gtv}')
print(f'Total GTV for retail partner bookings = ${retail_partner_gtv}')

print(
    f'\nTotal GTV for TV mounting for month based on bookings added = ${int(one_time_gtv + recurring_gtv + retail_partner_gtv)}')

Total GTV for 100 one-time mounting bookings = $13000
Total GTV for 30 handyman service bookings = $3120.0
Total GTV for retail partner bookings = $6750

Total GTV for TV mounting for month based on bookings added = $22870


### B

In [3]:
# Calculate cost of labor
pros_cost = 1.5 * 40

# One Time Bookings
discounted_price = 130 - 20

one_time_discounted_rev = (discounted_price - pros_cost)
one_time_rev = 130 - pros_cost

total_one_time_rev = (33 * one_time_discounted_rev) + ((100-33) * one_time_rev)

# Customers with recurring plans
recurring_rev = (130 - (130 * .2)) - pros_cost

total_recurring_rev = 30 * recurring_rev

# Customers with bookings through retail partner
retail_rev = (150 / 2) - pros_cost

total_retail_rev = 45 * retail_rev

# Print results including the total net revenue for the month
print(f'The total from one time bookings = ${int(total_one_time_rev)}')
print(
    f'The total revenue from recurring service bookings = ${int(total_recurring_rev)}')
print(f'The total revenue for retail bookings = ${int(total_retail_rev)}')

print(
    f'\nThe total revenue for the month = ${int(total_one_time_rev + total_recurring_rev + total_retail_rev)}')

The total from one time bookings = $6340
The total revenue from recurring service bookings = $1320
The total revenue for retail bookings = $675

The total revenue for the month = $8335


### C

- The one-time TV mounting bookings generate the highest revenue but also require the most resources since Angi has to find new customers each time. These bookings likely appeal to customers who only need a one-time service and don't want to commit to a recurring plan. The promotional coupons might suggest that these bookings require more marketing incentives to drive sales.


- Handyman service bookings yield less revenue per booking but offer the potential for greater long-term revenue, and recurring revenue as well as customer loyalty. These bookings likely appeal to customers who want to use multiple services and are interested in saving money with a discount. The discount rate of 20% may not be attractive enough for some customers to commit to this plan.


- Retail bookings have a lower net revenue since the retailer takes a 50% cut but is still a valuable channel for obtaining customers not familiar with their D2C offerings. The retail partner channel allows Angi to tap into an existing customer base and offer TV mounting services as an additional service.

Overall, it seems like Angi has various channels for acquiring TV-mounting customers, which could help mitigate risks associated with relying on one particular channel.

## Part 2: Business Performance Measurement

### A

In [4]:
# Idenitfy ROI

# Define variables for given metrics
eligible_skus = 6000000 * .4
expected_bookings = eligible_skus * .05
engineering_effort = 6
minimum_threshold_per_year = 3000

# Calculate minimum bookings required
min_bookings_required = minimum_threshold_per_year * engineering_effort

# Compute expected value of project
e_x = (0.30 * expected_bookings) + (0.70 * 0)

# Print Results
print(
    f'Expected bookings from project = {int(expected_bookings)} per year')
print(f'Minimum bookings required = {min_bookings_required} per year')

print(
    f'\nThe number of bookings from the project is {expected_bookings/min_bookings_required:.2f}x the minimum bookings required')

print(
    f"The expected value of the project is {e_x:.0f} bookings, and {e_x/min_bookings_required}x the minimum bookings required")

Expected bookings from project = 120000 per year
Minimum bookings required = 18000 per year

The number of bookings from the project is 6.67x the minimum bookings required
The expected value of the project is 36000 bookings, and 2.0x the minimum bookings required


Since the expected number of bookings is 2.0 times the minimum number of bookings required to justify the engineering effort, I would recommend Angi moves forward based on the calculated ROI. Although there's only a 30% chance of success the project should still be assessed further before moving forward.

#### Potential challenges

- Integration Challenges: Integrating Angi's booking platform with FurnitureCo's platform may require considerable engineering effort and time, which could cause delays and cost overruns.


- Customer Adoption: While the attach rate of 5% seems reasonable, it's not easy to predict how customers will respond to the offer. If there's a lack of interest in purchasing Angi's assembly & installation services then the project might not meet the expectations.


- Competing Offers: There could be other options for assembly and installation services, including competitors of Angi. If customers prefer other offers or services then the attach rate may be lower than expected.


- Market Saturation: The assembly and installation services market may be saturated, and there might not be enough demand for Angi's additional services.


- Retailer Commitment: The success of the project depends on FurnitureCo's commitment to promoting and offering Angi's assembly & installation services to its customers. If FurnitureCo doesn't actively promote Angi's services, the attach rate may also be lower than expected.


- Economic Uncertainty: Economic uncertainty could impact consumer spending and demand for assembly and installation services.

#### B

Start by calculating the expected value and cost for each price Tier $E(Revenue Per Booking)$ and $E(Cost Per Booking)$

Can do this by multiplying Angi's proposed price by the expected attach rate to calculate the expected revenue per booking and using the cost structure table to calculate the expected cost per booking

In [5]:
# Creating provided tables for easier calculation

# Creating projection table
tiers = ['$0.00-$99.99', '$100.00-$199.99', '$200.00-$299.99', '$300.00-$399.99',
         '$400.00-$499.99', '$500.00-$599.99', '$600.00-$799.99', '$800.00 (+)']
percent_skus = [.22, .10, .15, .6, .7, .10, .10, .20]
angi_price = [39, 59, 69, 89, 99, 109, 129, 179]

projection_df = pd.DataFrame({'Furniture Co Product Retail Price Tier': tiers,
                              '% Total Skus Sold': percent_skus, 'Proposed Angi Price': angi_price})


# Creating cost structure table
avg_hours = [2, 2, 2, 2, 2.5, 2.5, 3, 3]
labor_cost = [25 for i in range(8)]
variable_costs = [6 for i in range(8)]

cost_struct_df = pd.DataFrame({'Furniture Co Product Retail Price Tier': tiers,
                              'Average Hours': avg_hours,
                               'Hourly Labor Cost': labor_cost,
                               'Variable Cost': variable_costs})

display(projection_df)
display(cost_struct_df)

Furniture Co Product Retail Price Tier  % Total Skus Sold  \
0                           $0.00-$99.99               0.22   
1                        $100.00-$199.99               0.10   
2                        $200.00-$299.99               0.15   
3                        $300.00-$399.99               0.60   
4                        $400.00-$499.99               0.70   
5                        $500.00-$599.99               0.10   
6                        $600.00-$799.99               0.10   
7                            $800.00 (+)               0.20   

   Proposed Angi Price  
0                   39  
1                   59  
2                   69  
3                   89  
4                   99  
5                  109  
6                  129  
7                  179

Furniture Co Product Retail Price Tier  Average Hours  Hourly Labor Cost  \
0                           $0.00-$99.99            2.0                 25   
1                        $100.00-$199.99            2.0                 25   
2                        $200.00-$299.99            2.0                 25   
3                        $300.00-$399.99            2.0                 25   
4                        $400.00-$499.99            2.5                 25   
5                        $500.00-$599.99            2.5                 25   
6                        $600.00-$799.99            3.0                 25   
7                            $800.00 (+)            3.0                 25   

   Variable Cost  
0              6  
1              6  
2              6  
3              6  
4              6  
5              6  
6              6  
7              6

In [6]:
# Creating expected values

# Create table for expected values
expected_df = pd.DataFrame({'Furniture Co Product Retail Price Tier': tiers,
                           'Proposed Angi Price': angi_price, 'Attach Rate': [.05 for i in range(8)]})

# Creating copy for testing different prices
test_df = expected_df.copy()

# Create column for expected revenue per booking (multiplying proposed price by attach rate)
expected_df['E(Revenue Per Booking)'] = expected_df['Proposed Angi Price'] * \
    expected_df['Attach Rate']

# Create column for expected cost per booking (multiplying average hours and labor cost plus varibale costs by the attach rate)
expected_df['E(Cost Per Booking)'] = ((cost_struct_df['Average Hours'] *
                                      cost_struct_df['Hourly Labor Cost']) +
                                      cost_struct_df['Variable Cost']) * expected_df['Attach Rate']

# Create column for profit per booking (subtracting cost from revenue)
expected_df['E(Profit Per Booking)'] = expected_df['E(Revenue Per Booking)'] - \
    expected_df['E(Cost Per Booking)']

# Create column for profit margin (dividing net income by revenue)
expected_df['E(Profit Margin)'] = round((expected_df['E(Profit Per Booking)'] /
                                         expected_df['E(Revenue Per Booking)']) * 100, 2)

# Create column for weighted profit (multiplying profit margin by % of SKUs for each tier)
expected_df['E(Weighted Profit Margin)'] = expected_df['E(Profit Margin)'] * \
    projection_df['% Total Skus Sold']

# Display table
display(expected_df)

# Compute averages
avg_profit_margin = expected_df['E(Profit Margin)'].mean()
avg_weighted_profit_margin = expected_df['E(Weighted Profit Margin)'].mean()

# Print results
print(
    f'The expected average profit margin for the project = {avg_profit_margin:.2f}%')
print(
    f'The expected weighted average profit margin for the project = {avg_weighted_profit_margin:.2f}%')

Furniture Co Product Retail Price Tier  Proposed Angi Price  Attach Rate  \
0                           $0.00-$99.99                   39         0.05   
1                        $100.00-$199.99                   59         0.05   
2                        $200.00-$299.99                   69         0.05   
3                        $300.00-$399.99                   89         0.05   
4                        $400.00-$499.99                   99         0.05   
5                        $500.00-$599.99                  109         0.05   
6                        $600.00-$799.99                  129         0.05   
7                            $800.00 (+)                  179         0.05   

   E(Revenue Per Booking)  E(Cost Per Booking)  E(Profit Per Booking)  \
0                    1.95                2.800                 -0.850   
1                    2.95                2.800                  0.150   
2                    3.45                2.800                  0.650   
3                    4.45                2.800                  1.650   
4                    4.95                3.425                  1.525   
5                    5.45                3.425                  2.025   
6                    6.45                4.050                  2.400   
7                    8.95                4.050                  4.900   

   E(Profit Margin)  E(Weighted Profit Margin)  
0            -43.59                    -9.5898  
1              5.08                     0.5080  
2             18.84                     2.8260  
3             37.08                    22.2480  
4             30.81                    21.5670  
5             37.16                     3.7160  
6             37.21                     3.7210  
7             54.75                    10.9500

The expected average profit margin for the project = 22.17%
The expected weighted average profit margin for the project = 6.99%


<b>Angi <u>should propose a change</u> based on the expected weighted average profit margin of 6.99% as well as other challenges they may face in this project that would likely incur additional costs.</b> 

## How to achieve the 15% weighted profit margin


To reach the 15% profit margin prices would need to be significantly increased, and this would likely affect the attach rate as well.




Other ideas could be:

- Reduce costs: Angi could look for ways to reduce the costs of providing their services by negotiating better deals with partnerships or reducing labor costs.

- Increase sales volume: If Angi could increase the volume of services they provide, they may be able to benefit from economies of scale.

- Cross-sell and upsell: Angi could encourage customers to purchase additional services by cross-selling or upselling.

- Improve marketing and sales strategies: Angi could invest in marketing and sales strategies to increase awareness of products and the attach rate.

- Optimize pricing: Look for additional ways to optimize the pricing strategy.


In [7]:
# Test cell for looking at different price points and dropping Angi's services for the first tier

test_df['Proposed Angi Price'] = [55, 79, 99, 129, 149, 179, 209, 229]

test_df['E(Revenue Per Booking)'] = test_df['Proposed Angi Price'] * \
    test_df['Attach Rate']

test_df['E(Cost Per Booking)'] = ((cost_struct_df['Average Hours'] *
                                   cost_struct_df['Hourly Labor Cost']) +
                                  cost_struct_df['Variable Cost']) * test_df['Attach Rate']

test_df['E(Profit Per Booking)'] = test_df['E(Revenue Per Booking)'] - \
    test_df['E(Cost Per Booking)']

test_df['E(Profit Margin)'] = round((test_df['E(Profit Per Booking)'] /
                                     test_df['E(Revenue Per Booking)']) * 100, 2)

test_df['E(Weighted Profit Margin)'] = test_df['E(Profit Margin)'] * \
    projection_df['% Total Skus Sold']

display(test_df)

avg_weighted_profit_margin = test_df['E(Weighted Profit Margin)'].loc[1:].mean(
)

print(avg_weighted_profit_margin)

Furniture Co Product Retail Price Tier  Proposed Angi Price  Attach Rate  \
0                           $0.00-$99.99                   55         0.05   
1                        $100.00-$199.99                   79         0.05   
2                        $200.00-$299.99                   99         0.05   
3                        $300.00-$399.99                  129         0.05   
4                        $400.00-$499.99                  149         0.05   
5                        $500.00-$599.99                  179         0.05   
6                        $600.00-$799.99                  209         0.05   
7                            $800.00 (+)                  229         0.05   

   E(Revenue Per Booking)  E(Cost Per Booking)  E(Profit Per Booking)  \
0                    2.75                2.800                 -0.050   
1                    3.95                2.800                  1.150   
2                    4.95                2.800                  2.150   
3                    6.45                2.800                  3.650   
4                    7.45                3.425                  4.025   
5                    8.95                3.425                  5.525   
6                   10.45                4.050                  6.400   
7                   11.45                4.050                  7.400   

   E(Profit Margin)  E(Weighted Profit Margin)  
0             -1.82                    -0.4004  
1             29.11                     2.9110  
2             43.43                     6.5145  
3             56.59                    33.9540  
4             54.03                    37.8210  
5             61.73                     6.1730  
6             61.24                     6.1240  
7             64.63                    12.9260

15.203357142857143
